pruning/types/options.go

```diff
@@ -0,0 +1,135 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+)
+
+// PruningOptions defines the pruning strategy used when determining which
+// heights are removed from disk when committing state.
+type PruningOptions struct {
+	// KeepRecent defines how many recent heights to keep on disk.
+	KeepRecent uint64
+
+	// Interval defines when the pruned heights are removed from disk.
+	Interval uint64
+
+	// Type defines the type of pruning strategy. See below for more information.
+	Type Type
```

Comment:
Apparently, there are several `types` packages in the project. 

E.g:
- `sdk "github.com/cosmos/cosmos-sdk/types"`
- `storeTypes "github.com/cosmos/cosmos-sdk/store/types`
- `pruningTypes "github.com/cosmos/cosmos-sdk/pruning/types"`

Most of the time, we need to import several `types` packages in a file so we end up renaming them. As a result the pruning types become `pruningTypes` consistently and we use it as `pruningTypes.Type`.

If we go with `PruningType`, then when we are using the renamed package, this would be a code smell - smurf naming. Basically, it is an anti-pattern for repeating the same names such as `pruningType.PruningType`.

However, you are right that if we import only the pruning types package, it will get confusing without a rename. It's a trade-off.

I don't have a strong opinion on this so I'm going to change this to `PruningType`. Just wanted to explain the reasoning behind the choice of the name

Reply:
I renamed `Type` to `PruningStrategy` to avoid `pruningTypes.PruningTypes`. Let me know what you think

Reply:
sounds great to me!